custom_components/netdaemon/climate.py

```diff
@@ -0,0 +1,211 @@
+    for entity in client.entities:
+        if entity.split(".")[0] == PLATFORM_CLIMATE:
+            LOGGER.debug("Adding %s", entity)
+            climate_entities.append(
+                NetDaemonClimateEntity(coordinator, entity.split(".")[1])
+            )
+
+    if climate_entities:
+        async_add_devices(climate_entities)
+
+
+class NetDaemonClimateEntity(NetDaemonEntity, ClimateEntity):
+    """NetDaemon ClimateEntity class."""
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        if not self.entity_id:
+            return 0
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_SUPPORTED_FEATURES, 0)
+        )
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode.
+
+        Need to be one of HVAC_MODE_*.
+        """
+        if not self.entity_id:
+            return "off"
+        return self._coordinator.data[self.entity_id].get(ATTR_STATE, "off")
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        if not self.entity_id:
+            return 0.0
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_CURRENT_TEMPERATURE, 0.0)
+        )
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        if not self.entity_id:
+            return 0.0
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_TARGET_TEMPERATURE, 0.0)
+        )
+
+    @property
+    def target_humidity(self) -> int:
+        """Return the humidity we try to reach."""
+        if not self.entity_id:
+            return 0
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_TARGET_HUMIDITY, 0)
+        )
+
+    @property
+    def hvac_modes(self) -> list[str]:
+        """Return the list of available hvac operation modes.
+
+        Need to be a subset of HVAC_MODES.
+        """
+        if not self.entity_id:
+            return ["off"]
+
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_HVAC_MODES, [])
+        )
+
+    @property
+    def fan_modes(self) -> list[str]:
+        """Return the list of available fan modes.
+
+        Requires SUPPORT_FAN_MODE.
+        """
+        if not self.entity_id:
+            return ["off"]
+
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_FAN_MODES, [])
```

Comment:
```suggestion
            .get(ATTR_FAN_MODES, ["off"])
```
Maybe add "off" here to have something